### PR TITLE
Fix tests

### PR DESCRIPTION
**Pull request by skarimo**
- Update CI to use tox
- remove unneeded dependencies
- Fix tox 
- Disable some tests from running in non-replay mode

Some tests are skipped until I introduce proper resource cleanup.

### CODE REVIEW

File: tests/integration/resources/test_monitors.py [hunk 1]
```diff
@@ -2,11 +2,16 @@
...
 # Copyright 2019 Datadog, Inc.
+import pytest
 
+from tests.conftest import get_record_mode
 from tests.integration.helpers import BaseResourcesTestClass
 from datadog_sync.models import Monitors
 
 
+@pytest.mark.skipif(
+    get_record_mode() != "none", reason="Cannot create duplicate monitors. Reintroduce with  proper cleanup."
```
Comment (NouemanKHAL) [inline tests/integration/resources/test_monitors.py:13]:
```suggestion
    get_record_mode() != "none", reason="Cannot create duplicate monitors. Reintroduce with proper cleanup."
```

File: pytest.ini [hunk 2]
```diff
@@ -1,5 +1,3 @@
 [pytest]
 markers =
     integration: marks integration tests that require source/destination api+app keys.
-env =
-    MAX_WORKERS=1
```
Comment (NouemanKHAL) [inline pytest.ini:5]: Don't we need this for `pytest-recording` ?

Comment (skarimo) [reply pytest.ini:5]: Using tox now to set this and run tests so no need for this extra setting + dep

Comment (NouemanKHAL) [reply pytest.ini:5]: Yes, just thought maybe we can keep it for safety, it's up to you.

Comment (skarimo) [reply pytest.ini:5]: Ill omit for now and. reintroduce if needed later on

**Pull request merged**